python_modules/libraries/dagster-postgres/dagster_postgres/resources.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any, Optional
+
+import psycopg2
+from dagster import ConfigurableResource, InitResourceContext
+from dagster._utils.backoff import backoff
+from pydantic import Field, PrivateAttr
+
+
+class PostgresResource(ConfigurableResource):
+    """Resource for interacting with a postgres database. Wraps an underlying psycopg2 connection.
+
+    Note that the underlying pyscopg2.extensions.connection context-manager does
```

Comment:
This will also need to be reworded once we move connection open/close to happen in the context manager.